src/software/backend/replay_backend.cpp

```diff
@@ -0,0 +1,77 @@
+#include "replay_backend.h"
+
+#include <cstdlib>
+
+#include "software/util/design_patterns/generic_factory.h"
+
+ReplayBackend::ReplayBackend(const std::string& replay_input_dir)
+    : Backend(),
```

Comment:
nit: I don't think the `Backend` constructor needs to be called

Reply:
removed default constructors here